config/config_test.go

```diff
@@ -0,0 +1,110 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestConfigParser(t *testing.T) {
+	Convey("Parser should accept valid string", t, func() {
+		cfg, err := ParseConfig(configString)
+		So(cfg, ShouldNotBeNil)
+		So(err, ShouldBeNil)
+		validateConfig(cfg)
+	})
+
+	Convey("Parser should reject invalid strings - no listener config", t, func() {
+		cfg, err := ParseConfig(`
+			# no listener config
+			vault {
+				address       = "http://127.0.0.8200"
+			}
+			`)
+		So(cfg, ShouldBeNil)
+		So(err, ShouldNotBeNil)
+	})
+
+	Convey("Parser should reject invalid strings - no vault config", t, func() {
+		cfg, err := ParseConfig(`
+			listener "tcp" {
+				address       = "127.0.0.1:8000"
+			}
+			# no vault config
+			`)
+		So(cfg, ShouldBeNil)
+		So(err, ShouldNotBeNil)
+	})
+
+	Convey("Starting up a dev vault", t, func() {
+		cfg, shutdownCh, secretID, err := LoadConfigDev()
+		validateConfig(cfg)
+		So(shutdownCh, ShouldNotBeNil)
+		defer close(shutdownCh)
+		So(secretID, ShouldNotBeNil)
+		So(err, ShouldBeNil)
+		validateVaultHealth(cfg)
+	})
+
+	Convey("Loading custom config", t, func() {
+		cfg, err := LoadConfigFile("sample.hcl")
+		validateConfig(cfg)
+		So(err, ShouldBeNil)
+	})
+}
+
+func validateConfig(cfg *Config) {
```

Comment:
This could probably be simplified by using `ShouldResemble` https://github.com/smartystreets/goconvey/wiki/Assertions

E.g. 
```golang
So(cfg, ShouldResemble, Config{
		Listener: &ListenerConfig{
			Type:        "tcp",
			Address:     "127.0.0.1:8000",
			Tls_disable: true,
		},
		Vault: &VaultConfig{
			Type:           "vault",
			Address:        u.String(),
			Runtime_config: "secret/goldfish",
			Approle_login:  "auth/approle/login",
			Approle_id:     "goldfish",
		},
	})
```
To reuse this, define a const value equal to that config literal, and compare it to that everytime

I highly prefer unit tests to be as 'unit' as possible, and not to rely on calling helper functions unless really needed. Goconvey helps a lot in this regard :)